cloudevents-sdk-rdkafka/src/kafka_consumer_record.rs

```diff
@@ -0,0 +1,281 @@
+use crate::headers;
+use cloudevents::event::SpecVersion;
+use cloudevents::message::{
+    BinaryDeserializer, BinarySerializer, Encoding, MessageAttributeValue, MessageDeserializer,
+    Result, StructuredDeserializer, StructuredSerializer,
+};
+use cloudevents::{message, Event};
+use rdkafka::message::{BorrowedMessage, Headers, Message, OwnedMessage};
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::str;
+
+pub struct ConsumerRecordDeserializer {
+    pub(crate) headers: HashMap<String, Vec<u8>>,
+    pub(crate) payload: Vec<u8>,
+}
+
+impl ConsumerRecordDeserializer {
+    pub fn owned_new(message: OwnedMessage) -> ConsumerRecordDeserializer {
+        let mut resp_des = ConsumerRecordDeserializer {
+            headers: HashMap::new(),
+            payload: Vec::new(),
+        };
+        let headers = message.headers().unwrap();
+        for i in 0..headers.count() {
```

Comment:
Can you make a function to use that loop? notice that both `OwnedHeaders` and `BorrowedHeaders` implements `Headers` trait, so you can make a function that gets `impl Headers` and return the hashmap of headers

Reply:
I tried that a while back and it didnt work cause the type `&BorrowedMessage` doesn't implement `Headers`

Reply:
ok so now im taking reference for both, and am able to do something similar